receiver/otlpreceiver/go.mod

```diff
@@ -25,7 +25,6 @@ require (
 )
 
 require (
-	cloud.google.com/go/compute/metadata v0.2.4-0.20230617002413-005d2dfb6b68 // indirect
```

Comment:
Not sure what is wrong, but every time I run `make gotidy` I have this error:
```
go.opentelemetry.io/collector/exporter/otlpexporter imports
        go.opentelemetry.io/collector/config/configgrpc imports
        go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc tested by
        go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.test imports
        google.golang.org/grpc/interop imports
        golang.org/x/oauth2/google imports
        cloud.google.com/go/compute/metadata: ambiguous import: found package cloud.google.com/go/compute/metadata in multiple modules:
        cloud.google.com/go v0.26.0 (/Users/omrozowicz/go/pkg/mod/cloud.google.com/go@v0.26.0/compute/metadata)
        cloud.google.com/go/compute/metadata v0.2.4-0.20230617002413-005d2dfb6b68 (/Users/omrozowicz/go/pkg/mod/cloud.google.com/go/compute/metadata@v0.2.4-0.20230617002413-005d2dfb6b68)
make: *** [tidy] Error 1
```
For both otlp receiver and otlp exporter. When I bring back the changes from `main`, `make gotidy` deletes this automatically.

I inserted this line to make CI green but I am sure it's not a way to go:
```
replace cloud.google.com/go => cloud.google.com/go v0.110.2
```

Reply:
`replace cloud.google.com/go => cloud.google.com/go v0.110.2` is good. We have it all around the codebase